### PR TITLE
Release notes and updates for Mx4PC 2.10.0 (expected to be released on January 5)

### PR DESCRIPTION
**Pull request by zlogic**
Updated documentation and release notes for the new Mendix for Private Cloud release.

### CODE REVIEW

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md [hunk 1]
```diff
@@ -57,13 +55,50 @@ The following table lists the properties used as keys for database and storage-r
...
+
+To set a Mendix app constant, use the `mx-const-{name}` format (replace `{name}` with the name of the app constant).
+For example, if you need to set the `MyFirstModule.WelcomePageTitle` constant, specify its value via the `mx-const-MyFirstModule.WelcomePageTitle` key.
+
+To set a [Mendix Runtime custom setting](/refguide/custom-settings/), use the `mx-runtime-{name}` format (replace `{name}` with the name of the custom setting).
+For example, if you need to set the `com.mendix.storage.s3.EncryptionKeys` constant, specify its value via the `mx-runtime-com.mendix.storage.s3.EncryptionKeys` key.
+
+{{% alert color="info" %}}
+Most of the Mendix Runtime settings don't contain private data or are managed by the Mendix Operator, and overriding some values could lead to unexpected behaviour.
+Consult with Mendix Support to check if using CSI Secrets Storage for your use case would be considered a supported use case.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:93]: Have you confirmed with Support that they can deal with these queries?

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-deploy.md [hunk 2]
```diff
@@ -641,6 +643,29 @@ To add this setting:
...
+```bash {linenos=false}
+mxpc-cli log-extract -n {namespace} -f {filename}
+```
+
+This file can be shared with Mendix Support or the team responsible for maintaining infrastructure.
+
+{{% alert color="warning" %}}
+Before sending logs to support, please make sure that there they don't contain any sensitive or restricted information.
+
+To provide enough detail to work on an issue without disclosing sensitive information, you may want to redact some of the information in the log, or only keep messages from a specific time period.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-deploy.md:666]: Is this what you meant in your original. Or have I misunderstood?

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md [hunk 3]
```diff
@@ -57,13 +55,50 @@ The following table lists the properties used as keys for database and storage-r
...
+If your app is created in Mendix version 9.20 or above, and its Kubernetes service account is linked to an AWS IAM Role, you don't need to specify `storage-access-key-id` or `storage-secret-access-key` to access an S3 bucket.
+Instead, the same AWS IAM role can be used for S3 authentication.
+For more information and a complete walkthrough example, see the AWS Secrets Manager [example](#configure-using-aws-secrets-manager).
+{{% /alert %}}
+
+To set a Mendix app constant, use the `mx-const-{name}` format (replace `{name}` with the name of the app constant).
+For example, if you need to set the `MyFirstModule.WelcomePageTitle` constant, specify its value via the `mx-const-MyFirstModule.WelcomePageTitle` key.
+
+To set a [Mendix Runtime custom setting](/refguide/custom-settings/), use the `mx-runtime-{name}` format (replace `{name}` with the name of the custom setting).
+For example, if you need to set the `com.mendix.storage.s3.EncryptionKeys` constant, specify its value via the `mx-runtime-com.mendix.storage.s3.EncryptionKeys` key.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:87]: Should we link to the examples below, which make things clearer?

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md [hunk 4]
```diff
@@ -377,16 +420,50 @@ To enable your environment to use [AWS Secrets Manager](https://aws.amazon.com/b
...
 
 * It is not currently possible to use Storage Plans with CSI Secrets Storage. Instead, your infrastructure admin must attach a Kubernetes `ServiceAccount` and `SecretProviderClass` would be attached to the app before or after the environment is created.
-* If a secret is rotated or updated, the app should be restarted to use new credentials.
+* If a secret is rotated or updated, Mendix Operator version 2.10.0 and above will detect the changes and apply them. However (depending on the Mendix Runtime version in use), the changes might not be applied — if the changes are not applied, you should restart the environment. Bear in mind the following features of detecting a changed secret:
+    * There is a delay of a few minutes before the CSI Secrets Storage driver detects the changes.
+    * Only file storage credentials and `MxAdmin` password changes are correctly processed at the moment.
+    * In Mendix version 9.21 or below, rotation of database credentials requires the environment to be manually restarted. Rotation of database credentials will be added soon to the Mendix Runtime, and this document will be updated accordingly.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:468]: I updated this a bit as it was unclear. Have I changed the meaning?

**Review by MarkvanMents — APPROVED** [review]
I've made a few proofreading changes, but it looks good.
Please let me know when you publish this - Nidhi says that it may not be 5 January?

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-deploy.md (hunk 2 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-deploy.md:666]: Sometimes customers send complete logs without checking, and they contain too much information like passwords  and API keys. I wanted to remind customers that it's a good idea to quickly check the logs before sending them - especially if the logs are from the production environment.

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md (hunk 3 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:87]: I'll provide an example a bit later today - to make sure it actually works.

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md (hunk 1 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:93]: Not yet, but I expect those tickets to be forwarded directly to our team.

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md (hunk 4 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:468]: Looks good, I think it's clearer now.

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md (hunk 3 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:87]: My mistake - I misread the comment, I've added a link to the complete example below.

Comment (katarzyna-koltun-mx) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:87]: Thank you!

**Comment by katarzyna-koltun-mx** [comment]
@zlogic Hey Dmitrii, I'm back from PTO so I'll be publishing this PR when ready. Do we still want to publish tomorrow?

@MarkvanMents thank you for your help while I was away!

**Comment by zlogic** [comment]
@katarzyna-koltun-mx Yes, I think the plan to publish the new release tomorrow is still in place.

**Comment by katarzyna-koltun-mx** [comment]
@zlogic Great! I'll ping you to confirm tomorrow, and I'll publish once you give the go ahead. Thanks!

**Pull request merged**